### PR TITLE
Granular CRN for shared processor pool data sources and resources

### PR DESCRIPTION
**Pull request by Alexander-Kita**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

pi_shared_processor_pool (d):

```
=== RUN   TestAccIBMPIPISharedProcessorPoolDataSource_basic
--- PASS: TestAccIBMPIPISharedProcessorPoolDataSource_basic (26.84s)
PASS
```

pi_shared_processor_pools (d):

```
=== RUN   TestAccIBMPISharedProcessorPoolsDataSourceBasic
--- PASS: TestAccIBMPISharedProcessorPoolsDataSourceBasic (11.94s)
PASS
```

pi_shared_processor_pool (r):

```
=== RUN   TestAccIBMPISPPbasic
--- PASS: TestAccIBMPISPPbasic (49.45s)
PASS
```

### CODE REVIEW

File: ibm/service/power/data_source_ibm_pi_shared_processor_pool.go [hunk 1]
```diff
@@ -42,6 +42,11 @@ func DataSourceIBMPISharedProcessorPool() *schema.Resource {
 				Description: "The available cores in the shared processor pool.",
 				Type:        schema.TypeFloat,
 			},
+			Attr_CRN: {
+				Computed:    true,
+				Description: "CRN of shared processor pool.",
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_shared_processor_pool.go:47]:
```suggestion
				Description: "CRN of the shared processor pool.",
```

File: ibm/service/power/resource_ibm_pi_shared_processor_pool.go [hunk 2]
```diff
@@ -193,6 +205,9 @@ func resourceIBMPISharedProcessorPoolCreate(ctx context.Context, d *schema.Resou
 	if pg, ok := d.GetOk(Arg_SharedProcessorPoolPlacementGroupID); ok {
 		body.PlacementGroupID = pg.(string)
 	}
+	if tags, ok := d.GetOk(Arg_UserTags); ok {
+		body.UserTags = flex.ExpandStringList(tags.([]interface{}))
+	}
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_shared_processor_pool.go:210]: Could add a check if the Len()> 0 before setting it?

File: ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccIBMPISPPbasic(t *testing.T) {
+	name := fmt.Sprintf("tfpispp%d", acctest.RandIntRange(10, 100))
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go:23]:
```suggestion
	name := fmt.Sprintf("tf-pi-spp-%d", acctest.RandIntRange(10, 100))
```
For easy of readability

File: ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+					resource.TestCheckResourceAttr(sppRes, "pi_shared_processor_pool_name", name),
+					resource.TestCheckResourceAttrSet(sppRes, "crn"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccIBMPISPPusertags(t *testing.T) {
+	name := fmt.Sprintf("tfpispp%d", acctest.RandIntRange(10, 100))
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go:43]: Same as above

**Review by michaelkad — CHANGES_REQUESTED** [review]
Left few comments

File: ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go (hunk 3 above)
Comment (Alexander-Kita) [reply ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go:23]: I cannot add hyphens, but I can add underscores.

File: website/docs/r/pi_shared_processor_pool.html.markdown [hunk 5]
```diff
@@ -58,13 +58,15 @@ Review the argument references that you can specify for your resource.
...
 * `pi_shared_processor_pool_placement_group_id` - (Optional, String) The ID of the placement group the shared processor pool is created in.
+* `pi_user_tags` - (Optional, List) The user tags associated with this resource..
 
 ## Attribute reference
 
  In addition to all argument reference list, you can access the following attribute reference after your resource is created.
 
 * `allocated_cores` - (Float) The allocated cores in the shared processor pool.
 * `available_cores` - (Integer) The available cores in the shared processor pool.
+* `crn` - (String) The CRN of this resource..
```
Comment (ismirlia) [inline website/docs/r/pi_shared_processor_pool.html.markdown:69]:
```suggestion
* `crn` - (String) The CRN of this resource.
```

File: website/docs/r/pi_shared_processor_pool.html.markdown [hunk 6]
```diff
@@ -58,13 +58,15 @@ Review the argument references that you can specify for your resource.
 * `pi_shared_processor_pool_name` - (Required, String) The name of the shared processor pool.
 * `pi_shared_processor_pool_reserved_cores` - (Required, Integer) The amount of reserved cores for the shared processor pool.
 * `pi_shared_processor_pool_placement_group_id` - (Optional, String) The ID of the placement group the shared processor pool is created in.
+* `pi_user_tags` - (Optional, List) The user tags associated with this resource..
```
Comment (ismirlia) [inline website/docs/r/pi_shared_processor_pool.html.markdown:61]:
```suggestion
* `pi_user_tags` - (Optional, List) The user tags associated with this resource.
```

**Review by ismirlia — APPROVED** [review]

**Review by ismirlia — APPROVED** [review]

File: ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+			continue
+		}
+		cloudInstanceID, sppID, err := splitID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		sppC := instance.NewIBMPISharedProcessorPoolClient(context.Background(), sess, cloudInstanceID)
+		spp, err := sppC.Get(sppID)
+		if err == nil {
+			log.Println("shared processor pool*****")
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go:58]: You might want to remove this

Comment (Alexander-Kita) [reply ibm/service/power/resource_ibm_pi_shared_processor_pool_test.go]: Removed in next commit.

**Review by michaelkad — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
Opened in main repo

**Pull request closed without merging**